use nix shell for nix-update-* targets, fix nix-cache

jakubgs opened the pull request:
Changes:

* Remove `SHELL := /bin/sh` from definition of Makefile targets that update nix packages
* Check for Nix shell in those scripts and make `make shell` required for them
* Remove unnecessary command from `ci/Jenkinsfile.nix-cache`
* Add installation of `node2nix` if missing in `nix/desktop/realm-node/generate-nix.sh`
* Adjust the `find` for `nix-copy-closure` to use `-type d` which means we can't hit `*.lock`
* Changed `shell.nix` to load only most bare bones shell when `TARGET_OS=none`
* Changed `ci/Jenkinsfile.nix-cache` to use `TARGET_OS=none` by default.
* Made `make jsbundle-android` run in a pure shell in `ci/Jenkinsfile.nix-cache`

jakubgs committed:
use nix shell for nix-update-* targets

Signed-off-by: Jakub Sokołowski <jakub@status.im>

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 803fa600 | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-8652/1/) | 2019-07-26 20:47:53 | ~1 min | `ios` | [:page_facing_up: log](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-8652/1/consoleText) |

jakubgs commented:
```
invalid output dir: mkdir /private/var/folders/sn: permission denied
```
Ugh, I need to figure out what this one is about.

pedropombeiro commented on Makefile:
```diff
@@ -60,19 +60,15 @@ nix-purge: SHELL := /bin/sh
 nix-add-gcroots: ##@nix Add Nix GC roots to avoid status-react expressions being garbage collected
 	scripts/add-nix-gcroots.sh
 
-nix-update-npm: SHELL := /bin/sh
```
Should we add a specific Nix shell target for these Make targets so we don't force a ton of stuff to be downloaded/built?

jakubgs replied:
You're right.

pedropombeiro reviewed (approved):
Looks good, just have a minor concern about instantiating a heavy shell environment when the actual required dependencies are tiny.

jakubgs commented:
I've included changes from @PombeirP [comments in #8574](ttps://github.com/status-im/status-react/pull/8574#discussion_r308587282):

* Changed `shell.nix` to load only most bare bones shell when `TARGET_OS=none`
* Changed `ci/Jenkinsfile.nix-cache` to use `TARGET_OS=none` by default.
* Made `make jsbundle-android` run in a pure shell in `ci/Jenkinsfile.nix-cache`
